pkg/client/testing/core/fixture.go

```diff
@@ -286,6 +297,15 @@ func (t *tracker) add(obj runtime.Object, replaceExisting bool) error {
 			return err
 		}
 
+		// Propagate namespace to the new object if hasn't already been set.
+		if len(newMeta.GetNamespace()) == 0 {
+			newMeta.SetNamespace(ns)
+		}
+
+		if newMeta.GetNamespace() != ns {
```

Comment:
@soltysh added a commit to populate the error with information about the mismatched namespaces. Thanks for reviewing.